roles/openshift_logging_eventrouter/tasks/main.yaml

```diff
@@ -0,0 +1,88 @@
+# allow passing in a tempdir
+- name: Create temp directory for doing work in
+  command: mktemp -d /tmp/openshift-logging-ansible-XXXXXX
+  register: mktemp
+  changed_when: False
+
+- set_fact:
+    tempdir: "{{ mktemp.stdout }}"
+
+- name: Create templates subdirectory
+  file:
+    state: directory
+    path: "{{ tempdir }}/templates"
+    mode: 0755
+  changed_when: False
+
+
+# create service account
+- name: Create EventRouter service account
+  oc_serviceaccount:
+    state: present
+    name: "aggregated-logging-eventrouter"
+    namespace: "{{ openshift_logging_eventrouter_namespace }}"
+
+- name: Create event-reader cluster role
+  oc_clusterrole:
+    state: present
+    name: event-reader
+    rules:
+    - apiGroups:
+      - ""
+      resources:
+      - events
+      verbs:
+      - get
+      - watch
+      - list
+
+# set service account permissions
+- name: Set privileged permissions for EventRouter
+  oc_adm_policy_user:
+    namespace: "{{ openshift_logging_eventrouter_namespace }}"
+    resource_kind: cluster-role
+    resource_name: event-reader
+    state: present
+    user: "system:serviceaccount:{{ openshift_logging_eventrouter_namespace }}:aggregated-logging-eventrouter"
+
+
+# create EventRouter config
+- name: Generate EventRouter DC template
+  template:
+    src: eventrouter-config.j2
+    dest: "{{ tempdir }}/templates/config.json"
+  vars:
+    sink: "{{ openshift_logging_eventrouter_sink }}"
+
+# set configmap
+- name: Set EventRouter configmap
+  oc_configmap:
+    state: present
+    name: logging-eventrouter
+    namespace: "{{ openshift_logging_eventrouter_namespace }}"
+    from_file:
+        config.json: "{{ tempdir }}/templates/config.json"
+
+# create EventRouter deployment config
+- name: Generate EventRouter DC template
+  template:
+    src: eventrouter-dc.j2
+    dest: "{{ tempdir }}/templates/eventrouter-dc.yaml"
+  vars:
+    replicas: "{{ openshift_logging_eventrouter_replicas }}"
+    node_selector: "{{ openshift_logging_eventrouter_nodeselector | default({}) }}"
+    cpu_limit: "{{ openshift_logging_eventrouter_cpu_limit }}"
+    memory_limit: "{{ openshift_logging_eventrouter_memory_limit }}"
+
+
+- name: Set EventRouter DC
+  oc_obj:
+    state: present
+    name: logging-eventrouter
+    namespace: "{{ openshift_logging_eventrouter_namespace }}"
+    kind: dc
+    files:
+    - "{{ tempdir }}/templates/eventrouter-dc.yaml"
+    delete_after: false
+
+
```

Comment:
I dont see it hear but I believe in the logging and metrics role we have a task that deletes the tempdir we created initially.